### PR TITLE
Add GitHub CI with loose pre-commit

### PR DESCRIPTION
**Pull request by rodrigodelazcano**
Initial PR to include the GitHub CI, very similar to Gym's. The loose pre-commit means that flake8 and pyright are excluding all of the files to pass the tests. Future PR's while make these tests available once the testing errors are fixed.

The PR also includes the auto-corrections made by `black`.

### CODE REVIEW

File: benchmark.py [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 
-import time
 import argparse
-import gym_minigrid
+import time
+
 import gym
+
+import gym_minigrid
 from gym_minigrid.wrappers import *
```
Comment (pseudo-rnd-thoughts) [inline benchmark.py:9]: We should remove the * and replace with the actual stuff

File: gym_minigrid/envs/fetch.py [hunk 2]
```diff
@@ -55,28 +52,30 @@ def _gen_grid(self, width, height):
...
         self.targetColor = target.color
 
-        descStr = '%s %s' % (self.targetColor, self.targetType)
+        descStr = f"{self.targetColor} {self.targetType}"
 
         # Generate the mission string
         idx = self._rand_int(0, 5)
         if idx == 0:
-            self.mission = 'get a %s' % descStr
+            self.mission = "get a %s" % descStr
```
Comment (pseudo-rnd-thoughts) [inline gym_minigrid/envs/fetch.py:60]: format string?

File: gym_minigrid/envs/gotodoor.py [hunk 3]
```diff
@@ -60,7 +58,7 @@ def _gen_grid(self, width, height):
         self.target_color = doorColors[doorIdx]
 
         # Generate the mission string
-        self.mission = 'go to the %s door' % self.target_color
+        self.mission = "go to the %s door" % self.target_color
```
Comment (pseudo-rnd-thoughts) [inline gym_minigrid/envs/gotodoor.py:61]: format string

File: gym_minigrid/envs/lockedroom.py [hunk 4]
```diff
@@ -103,22 +88,19 @@ def _gen_grid(self, width, height):
...
-            size=(rWallIdx-lWallIdx, height)
+            top=(lWallIdx, 0), size=(rWallIdx - lWallIdx, height)
         )
 
         # Generate the mission string
         self.mission = (
-            'get the %s key from the %s room, '
-            'unlock the %s door and '
-            'go to the goal'
+            "get the %s key from the %s room, "
```
Comment (pseudo-rnd-thoughts) [inline gym_minigrid/envs/lockedroom.py:96]: format string

File: gym_minigrid/minigrid.py [hunk 5]
```diff
@@ -1,61 +1,56 @@
...
-import gym
+import math
 from enum import IntEnum
+
+import gym
 import numpy as np
 from gym import error, spaces, utils
 from gym.utils import seeding
+
 from .rendering import *
```
Comment (pseudo-rnd-thoughts) [inline gym_minigrid/minigrid.py:10]: Remove relative imports and *

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+include = [
+    "gym_minigrid/**",
+    "tests/**"
+]
+
+exclude = [
+    "**/node_modules",
+    "**/__pycache__",
+
+   "gym_minigrid/**",
```
Comment (pseudo-rnd-thoughts) [inline pyproject.toml:12]: So pyright is added but won't check anything?

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
This PR adds alot of things that could be separate PRs, one for the ISSUE TEMPLATE, one for build CI and one for pre-commit

**Comment by rodrigodelazcano** [comment]
Should I close this PR and make separates? Or continue in this one?

**Comment by rodrigodelazcano** [comment]
Closing this PR for this one https://github.com/Farama-Foundation/gym-minigrid/pull/200

**Pull request closed without merging**